kernel/src/path.rs

```diff
@@ -73,6 +75,8 @@ impl AsUrl for FileMeta {
 
 impl<Location: AsUrl> ParsedLogPath<Location> {
     // NOTE: We can't actually impl TryFrom because Option<T> is a foreign struct even if T is local.
+    #[cfg_attr(feature = "developer-visibility", visibility::make(pub))]
+    #[cfg_attr(not(feature = "developer-visibility"), visibility::make(pub(crate)))]
     pub fn try_from(location: Location) -> DeltaResult<Option<ParsedLogPath<Location>>> {
```

Comment:
```suggestion
   fn try_from(location: Location) -> DeltaResult<Option<ParsedLogPath<Location>>> {
```
(several more below)